src/Main.elm

```diff
@@ -1150,32 +1153,48 @@ viewFormFieldOptionsBuilder shortTextTypeList index formField =
                 ]
     in
     case formField.type_ of
-        ShortText inputType maybeMaxLength ->
+        ShortText inputType attrs ->
             let
+                attrsDict =
+                    Dict.fromList attrs
+
                 maybeShortTextTypeMaxLength =
                     shortTextTypeList
                         |> List.filter (\( k, _ ) -> k == inputType)
                         |> List.head
                         |> Maybe.map Tuple.second
                         |> Maybe.andThen (Dict.get "minlength")
                         |> Maybe.andThen String.toInt
+
+                otherAttrs =
+                    List.map (\( k, v ) -> Html.Attributes.attribute k v)
+                        attrs
             in
             [ case maybeShortTextTypeMaxLength of
                 Nothing ->
                     div [ class "tff-field-group" ]
                         [ label [ class "tff-field-label", for ("maxlength-" ++ idSuffix) ] [ text "Max length (optional)" ]
                         , input
-                            [ id ("maxlength-" ++ idSuffix)
-                            , type_ "number"
-                            , class "tff-text-field"
-                            , value (Maybe.map String.fromInt maybeMaxLength |> Maybe.withDefault "")
-                            , onInput (OnFormField OnMaxLengthInput index)
-                            ]
+                            ([ id ("maxlength-" ++ idSuffix)
+                             , type_ "number"
+                             , class "tff-text-field"
+                             , value (Dict.get "maxlength" attrsDict |> Maybe.withDefault "")
+                             , onInput (OnFormField OnMaxLengthInput index)
+                             ]
+                                ++ otherAttrs
+                            )
                             []
                         ]
 
                 Just i ->
-                    input [ type_ "hidden", name ("maxlength-" ++ idSuffix), value (String.fromInt i) ] []
+                    input
+                        ([ type_ "hidden"
+                         , name ("maxlength-" ++ idSuffix)
+                         , value (String.fromInt i)
+                         ]
+                            ++ otherAttrs
```

Comment:
just noticed this `otherAttrs` is for the `CollectData` mode, not for the form builder fields

```suggestion
```